libctru/source/applets/error.c

```diff
@@ -0,0 +1,74 @@
+#include <3ds.h>
+#include <stdlib.h>
+#include <string.h>
+#include <malloc.h>
+#include <3ds/types.h>
+#include <3ds/result.h>
+#include <3ds/svc.h>
+#include <3ds/synchronization.h>
+#include <3ds/services/apt.h>
+#include <3ds/ipc.h>
+#include <3ds/env.h>
+#include <3ds/util/utf.h>
+#include <3ds/applets/error.h>
+
+void error_Init(ErrConf* err, ErrorType type, err_lang lang)
+{
+	memset(err, 0, sizeof(*err));
+	err->errorType=         type;
+	err->useLanguage=       lang;
+	err->upperScreenFlag= NORMAL;
+	err->eulaVersion =         0;
+	err->homeButton  =      true;
+        err->softwareReset =   false;
+        err->appJump     =     false;
+        err-> returnCode   = UNKNOWN;
+}
+
+static void errorConvertToUTF16(u16* out, const char* in, int max)
+{
+	if (!in || !*in)
+	{
+		out[0] = 0;
+		return;
+	}
+
+	ssize_t units = utf8_to_utf16(out, (const uint8_t*)in, max);
+	if (units < 0)
+	{
+		out[0] = 0;
+		return;
+	}
+
+	out[units] = 0;
+}
+char k[1900];
+char *c_shift(char text[])
```

Comment:
Should be static. I'm not really sure what is the purpose of shifting the buffer by one and duplicating the first byte, but this would be equivalent and more readable:

```
static char* c_shift(const char *text)
{
	size_t len = strlen(text);
	char *str = (char*)malloc(len + 2);
	if(str)
	{
		str[0] = text[0];
		strcpy(&str[1], text);
	}
	return str;
}
```